src/webots/nodes/WbReceiver.cpp

```diff
@@ -177,6 +180,29 @@ void WbReceiver::updateTransmissionSetup() {
   mNeedToConfigure = true;
 }
 
+bool WbReceiver::isChannelAllowed() {
```

Comment:
This function should also be called in `WbReceiver::updateTransmissionSetup` to check if the new `Receiver.channel` value is valid and reset it if not.